### ISSUE TITLE
Not Thread-Safe

### ISSUE THREAD

**Issue by Meorge** [paddycarey/dweepy#4]
I'm trying to use dweepy for a program to create a Google Docs-style collaboration mode. However, it seems like I have to run it in a thread, which apparently is not good (it uses urllib/htmllib, which isn't thread-safe). Is there a way to port this to urllib3/htmllib2?

**Comment by paddycarey** [comment]
I'd certainly like to fix this, maybe you can help me understand the issue a bit better.

dweepy uses the `requests` library for all of its HTTP calls (which I thought built on `urllib3` but maybe i'm mistaken?). The only place I import from urllib is that i use the `quote` function when setting an alert.

Do you know what exactly is making the library unsafe?

**Comment by Meorge** [comment]
No, I don't think I know exactly what. Here's my traceback: http://pastebin.com/MJdRriyH
From that, it looks like it's httplib not working in urllib3, but I'm not sure (I'm a relative novice in Python)

**Comment by paddycarey** [comment]
Thanks for the traceback, I *think* I know what the issue is but i'll dig in more and hopefully should have a fix for this when i get some time over the next few days.

**Comment by Meorge** [comment]
Sounds great! Thanks for the quick response :smile:

**Comment by Meorge** [comment]
Do you know what the issue is?

**Comment by paddycarey** [comment]
Sorry about the delay, I'm looking at this again now. I thought I knew what the issue was but I'm having trouble recreating on my own rig (Linux/AMD64). I've tried the same python and requests versions that I can see in your traceback but still can't get it to fail.

Could you possibly provide a more complete example of exactly what's failing for you? Justa  quick function that I can run that shows the failing behaviour and I'll try to find a Mac to test it out on.

**Comment by Meorge** [comment]
I just tried to make a small program that would give the error again, but it's actually working fine :\
http://pastebin.com/Zc4S8g4K

Would I be able to send you the snippet of code in my program that's causing the error?
